Projects/Checkers/Board.cs

```diff
@@ -0,0 +1,171 @@
+
+	public Board()
+	{
+		Aggressor = null;
+		Pieces = new List<Piece>
+			{
+				new() { NotationPosition ="A3", Color = Black},
+				new() { NotationPosition ="A1", Color = Black},
+				new() { NotationPosition ="B2", Color = Black},
+				new() { NotationPosition ="C3", Color = Black},
+				new() { NotationPosition ="C1", Color = Black},
+				new() { NotationPosition ="D2", Color = Black},
+				new() { NotationPosition ="E3", Color = Black},
+				new() { NotationPosition ="E1", Color = Black},
+				new() { NotationPosition ="F2", Color = Black},
+				new() { NotationPosition ="G3", Color = Black},
+				new() { NotationPosition ="G1", Color = Black},
+				new() { NotationPosition ="H2", Color = Black},
+
+				new() { NotationPosition ="A7", Color = White},
+				new() { NotationPosition ="B8", Color = White},
+				new() { NotationPosition ="B6", Color = White},
+				new() { NotationPosition ="C7", Color = White},
+				new() { NotationPosition ="D8", Color = White},
+				new() { NotationPosition ="D6", Color = White},
+				new() { NotationPosition ="E7", Color = White},
+				new() { NotationPosition ="F8", Color = White},
+				new() { NotationPosition ="F6", Color = White},
+				new() { NotationPosition ="G7", Color = White},
+				new() { NotationPosition ="H8", Color = White},
+				new() { NotationPosition ="H6", Color = White}
+			};
+	}
+
+	public static string ToPositionNotationString(int x, int y)
+	{
+		if (!IsValidPosition(x, y)) throw new ArgumentException("Not a valid position!");
+		return $"{(char)('A' + x)}{y + 1}";
+	}
+
+	public static (int X, int Y) ParsePositionNotation(string notation)
+	{
+		if (notation is null) throw new ArgumentNullException(nameof(notation));
+		notation = notation.Trim().ToUpper();
+		if (notation.Length is not 2 ||
+			notation[0] < 'A' || 'H' < notation[0] ||
+			notation[1] < '1' || '8' < notation[1])
+			throw new FormatException($@"{nameof(notation)} ""{notation}"" is not valid");
+		return (notation[0] - 'A', notation[1] - '1');
+	}
+
+	public static bool IsValidPosition(int x, int y) =>
+		0 <= x && x < 8 &&
+		0 <= y && y < 8;
+
+	public (Piece A, Piece B) GetClosestRivalPieces(PieceColor priorityColor)
+	{
+		double minDistanceSquared = double.MaxValue;
+		(Piece A, Piece B) closestRivals = (null!, null!);
+		foreach (Piece a in Pieces.Where(piece => piece.Color == priorityColor))
+		{
+			foreach (Piece b in Pieces.Where(piece => piece.Color != priorityColor))
+			{
+				(int X, int Y) vector = (a.X - b.X, a.Y - b.Y);
+				double distanceSquared = vector.X * vector.X + vector.Y * vector.Y;
+				if (distanceSquared < minDistanceSquared)
+				{
+					minDistanceSquared = distanceSquared;
+					closestRivals = (a, b);
+				}
+			}
+		}
+		return closestRivals;
+	}
+
+	public List<Move> GetPossibleMoves(PieceColor color)
+	{
+		List<Move> moves = new();
+		if (Aggressor is not null)
+		{
+			if (Aggressor.Color != color)
+			{
+				throw new Exception();
```

Comment:
You should pass a string into the `Exception()` something meaningful, this exception won't tell you anything meaningful as it is now

Reply:
Idk, maybe like `throw new Exception("Aggressor.Color is not equal to color")` etc

Reply:
fixed in 9c14b10a5f41fd83ff6a135659a3f44a513c197c. I realize the text I added isn't that intuitive, but this exception shouldn't ever occur unless there is a code bug